libopae/plugins/xfpga/metrics/metrics_int.h

```diff
@@ -0,0 +1,217 @@
+// Copyright(c) 2018, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/**
+* \file metrics_int.h
+* \brief fpga metrics utils functions
+*/
+
+#ifndef __FPGA_METRICS_INT_H__
+#define __FPGA_METRICS_INT_H__
+
+#include "vector.h"
+#include "opae/metrics.h"
+
+// Power,Thermal & Performance definations
+
+#define IPERF                               "iperf"
+#define DPERF                               "dperf"
+
+#define PWRMGMT                             "power_mgmt"
+#define THERLGMT                            "thermal_mgmt"
+#define REVISION                            "revision"
+#define PERF_FREEZE                         "freeze"
+#define PERF_ENABLE                         "enable"
+#define PERF_CACHE                          "cache"
+#define PERF_FABRIC                         "fabric"
+#define PERF_IOMMU                          "iommu"
+#define PERF                                 "performance"
+#define FPGA_LIMIT                           "fpga_limit"
+#define XEON_LIMIT                           "xeon_limit"
+#define TEMP                                 "Centigrade"
+
+
+#define BMC_LIB                              "libbmc.so"
+
+// AFU DFH Struct
+struct DFH {
```

Comment:
If this isn't specific to metrics, can we place it in a more common location for re-use?